Improve error when PR does not have kind

When the release workflow happen, all inlcuded PRs are scanned and if they contain something on the "Release Note" block they are processed. The script `hack/get-next-version.go` expects every PR with release note to have a label starting with `kind`. If a PR contains a release note but not a kind label, it fails with a cryptic error:

```
+ go run hack/get-next-version.go --release-notes-file ./release-notes.json --current-version v0.7.0
panic: interface conversion: interface {} is nil, not []interface {}

goroutine 1 [running]:
main.main()
	/home/runner/work/kubearchive/kubearchive/hack/get-next-version.go:45 +0xa34
exit status 2
```

**Acceptance Criteria**

* Handle the error and output which is the PR that do not have kind label.

--- a/hack/get-next-version.go
+++ b/hack/get-next-version.go
@@ -4,6 +4,7 @@ package main
 
 import (
 	"encoding/json"
+	"errors"
 	"flag"
 	"fmt"
 	"os"
@@ -39,11 +40,22 @@ func main() {
 	}
 
 	var collectedKinds string
+	var errs error
 	for _, pr := range releaseNotes {
-		kinds := pr.(map[string]any)["kinds"].([]any)
-		for _, kind := range kinds {
+		prMap := pr.(map[string]any)
+		kinds, ok := prMap["kinds"]
+		if !ok {
+			errs = errors.Join(fmt.Errorf("PR '%v' does not have 'kind/*' labels, please add one.", prMap["pr_number"]), errs)
+			continue
+		}
+
+		for _, kind := range kinds.([]any) {
 			collectedKinds += fmt.Sprintf("%s ", kind.(string))
 		}
+	}
+
+	if errs != nil {
+		panic(fmt.Sprintf("some PRs do not have 'kind/*' labels:\n%s", errs.Error()))
 	}
 
 	fmt.Fprintf(os.Stderr, "DEBUG: Collected Kinds: %s\n", collectedKinds)

